src/mlpack/base.hpp

```diff
@@ -68,38 +71,6 @@
   #define mlpack_force_inline __forceinline
 #endif
 
-// detect C++17 mode
-#if (__cplusplus >= 201703L)
```

Comment:
Suggest to keep the C++17 detection parts (directly related to `__cplusplus >= 201703L` and `_MSVC_LANG >= 201703L)`, so that we end up with `MLPACK_HAVE_CXX17` defined.  This is because MSVC likes to be "special".

Suggest to change the `#if defined(MLPACK_HAVE_CXX17)` block to:
```
#if !defined(MLPACK_HAVE_CXX17)
  #error "Need to enable C++17 mode in your compiler"
#endif
```

Reply:
Done